### DISCUSSION TITLE
Any clues about updating old C# native modules?

### DISCUSSION THREAD

**Discussion by MacKenzieHnC** [microsoft/react-native-windows#11347] [Q&A] [answered]
I can add more details later when I'm at my computer, but I suspect this is a known issue.

Using older (~0.68) C# native modules, I'm getting a lot of issues like "duplicate assembly reference" or "duplicate runtime.something". It seems like this has to do with a major change in how UWP compiling happens.

Any documentation somewhere on how to properly upgrade? The best I could find were StackOverflow answers that said to just comment out offending code, but that feels like the wrong way.

Besides, that solution has so far only allowed me to run debug builds; release is still broken.

For an example, `react-native-fs` seems to be broken.

**Comment by chrisglein** [comment, marked as answer]
That module in particular has an unmerged PR that modernizes the implementation: https://github.com/itinance/react-native-fs/pull/952

That should show how to do this (more documentation on https://aka.ms/reactnative), and also just might have the implementation you're looking for.

**Reply by MacKenzieHnC** [reply]
Are you saying the solution is to ditch C# altogether like that PR does? It seemed from what I was reading it would be as simple as updating the assembly references or downgrading to VS 2019 (see the long comment on [this StackOverflow answer](https://stackoverflow.com/a/62455235/5859568)). Something about earlier VS versions using a list of what to compile whereas VS 2022 needs a list of what NOT to compile.

EDIT: The `react-native-fs` PR works, thank you. I am still curious if you're saying C# is currently not viable during this transition RNW seems to be going through.
